### ISSUE TITLE
[TouhouDB] New album types / change Drama PV song type to just "Drama"

### ISSUE THREAD

**Issue by saturclay** [VocaDB/vocadb#1641]
### Is your feature request related to a problem? Please describe. (required)
1. Currently, there are 9991 albums on TouhouDB, 9818 (98.2%) of which are in the "Fanmade/doujin" category. This makes the fanmade category basically useless, as it covers nearly the whole database. For reference, the largest album category on VocaDB accounts for less than half of the total number of albums.
2. There are only 18 songs in the "drama PV" category that have original media, whereas there are ~86 songs tagged "voice drama", which are spread across the "drama PV" and "other" song types.

### Describe the solution you'd like. (required)
1. Of the fanmade albums, 641 are tagged "single" (not counting short demo albums which could also be counted as singles, but which aren't currently tagged) and 1153 are tagged "EP". **I propose that the "Fanmade/doujin" category be changed to "Fanmade album," and three additional categories be added: "Fanmade single", "Fanmade E.P.", and "Drama".** The last category is much smaller at 19 albums, but it's such a distinct type of album that I think it warrants its own category.
2. I'd like to **change the name of the "drama PV" song type on TouhouDB to "drama"**, to accommodate both drama PVs and voice dramas, since voice dramas are much more common than drama PVs.

### Describe alternatives you've considered. (optional)
Alternatively, the song type could be changed to "voice drama" since those are more common, but I think it makes sense to keep the category broader.

### Checklist (required)

Fill out the checklist.

- [x] Searched **[VocaDB/vocadb](https://github.com/VocaDB/vocadb/issues)** for duplicate issues.

**Comment by saturclay** [comment]
Something that also occurs to me is that we (again, just talking about TouhouDB) could really use an additional song type for second-order arrangements (e.g. remasters, covers, etc. that are arrangements of other arrangements), because it can get really confusing when an arrangement is remixed/re-recorded/covered to determine which was the original arrangement... but I'm not sure what the best way to implement this would be because it's not always 100% clear whether a track is a remaster, or a re-recorded version, or a remix, especially for tracks where media isn't available.

**Comment by saturclay** [comment]
Given that no other TDB users have given me any feedback, here's my final proposal.

Album types:

- Original album
- Game
- Fanmade album (changed from Fanmade/doujin)
- Fanmade EP (new)
- Fanmade single (new)
- Other

Song types:

- Original song
- Arrangement
- Remix (new)
- Cover (new)
- Remaster (new)
- Short version (new)
- Instrumental
- Drama (changed from Drama PV)
- Music PV
- Other

**Comment by andreoda** [comment]
A few types that I see as questionable:

Album types:
- Fanmade album
-> isn't it better to just use a tag for those?
- Fanmade E.P.
-> if "Fanmade album" exists, looks redundant/like a duplicate
- Fanmade single
-> if "Fanmade album" exists, looks redundant/like a duplicate

Song types:
- Arrangement
-> isn't it the same as "Remix"?
- Short version
-> isn't it better to just use a tag for those? example: what if a song is an "original" but also a "short version" at the same time?

**Comment by saturclay** [comment]
The main reason I included the fanmade/original split is because that's the existing system; however, I think that eliminating that distinction would be fine. Rather than having a tag for "fanmade", the issue is that there are so few albums in the original category that it makes more sense to just have a tag for those (which already exists because all original albums are in one of [these](https://touhoudb.com/T/838) [two](https://touhoudb.com/T/629/touhou-ost) tags) and assume everything else is fanmade.

Regarding song types, I realized there's some additional context I didn't include here or in my [comment above](https://github.com/VocaDB/vocadb/issues/1641#issuecomment-1876199221). Here's what I said in the TDB discord:

> The hardest part would be just making it clear to people what the difference is between an arrangement and a remix. Here's a breakdown by number of entries:
> 
>* remix: ~5700
>* cover: ~1900 (actual number would be smaller because some things are tagged with both cover and remix, but i think remix would probably get precedence)
>* short ver: ~970
>* remaster: ~686
> 
> There are also 181 songs tagged "demo", which are a little bit hard to deal with, but I don't think it's crucial that there be a demo category since it's not quite so many songs (although that is larger than the music pv category at the moment).

As you can see, a large number of songs are remixes (of *other arrangements*) (and there's [already a tag for this](https://touhoudb.com/T/304/remix)). This, combined with the fact that it can get really confusing to immediately tell which is the "original" arrangement vs. a remix is what necessitates a category for remixes. With regard to short versions, I suggested that as its own category because of how many short versions there are, but I'd be fine with just keeping that as a tag and rolling those into the remix category. Either way, a new category is needed.

In an ideal world where all songs were clearly named (_ remix) or (short version), these things wouldn't cause me so much confusion, but I'll give a few examples of cases where a new type is badly needed.
* Let's say I want to add a remix of the song [U-N-K-N-O-W-N](https://touhoudb.com/S/17370). Here's what comes up when I select the parent version:
![Screenshot 2024-02-01 at 12 25 36 PM](https://github.com/VocaDB/vocadb/assets/142631139/1829a925-c0dc-4452-b779-a0343d341181)
Some of them are labeled, but some of them have the exact same title. In this case, the first one is the original arrangement, and the second two are short versions, but there's no way of telling without clicking through to each (and short versions/remixes aren't always added to the db before the original version).
* There are 19 songs right now called [Night of Nights](https://touhoudb.com/Search?searchType=Anything&childTags=false&draftsOnly=false&filter=%22night%20of%20nights%22&page=1&pageSize=100). Of course, anyone with much familiarity with Touhou music will know the BeatMARIO one is the original, but there are 17 other remixes which don't have "remix" in the title (because the remixer did not include it in their upload), and a short version which also has the same title.

I don't think we should have to add anything to the title of a song that wasn't there in the tracklist or upload if at all possible; at least, not to the primary name for a song. The only other solution I can think is to display the released date of each song when you're selecting an original, to make it clear which came first. But even then, I would still recommend having a dedicated "remix" category, because it feels like a significant distinction to make between "arrangements of ZUN originals" and "remixes of arrangements of ZUN originals". In any case, the onus will be on me to clearly document the difference between arrangements and remixes.

**Comment by saturclay** [comment]
Addendum: @andreoda Do you think calling it "rearrangement" instead of remix would be clearer? I think that could be a good compromise.

**Comment by andreoda** [comment]
@saturclay
That is okay too.
I think, in the end you need to decide that, since I am not as active on TDB.
You made a good analysis regarding this issue.
As long as you got proper reasons for these changes, thought about the benefits + drawbacks, I think it's are all good.

**Comment by saturclay** [comment]
Ok, real final proposal below (also can't believe I forgot to mention drama CDs before, since I mentioned that in my original post):

Album types:

- Game (ゲーム)
- Album (アルバム) [merges original album and doujin/fanmade]
- E.P. (E.P.) [new]
- Single (シングル) [new]
- Drama (ドラマ) [new; If the name needs to be unique from the song type, this can be Drama CD/ドラマCD]
- Other (その他)

Song types:

- Original song (オリジナル曲)
- Arrangement (編曲)
- Rearrangement (再編曲) [new]
- Cover (カバー曲) [new]
- Remaster (リマスター) [new]
- Short version (ショートバージョン) [new]
- Instrumental (インストゥルメンタル)
- Drama (ドラマ) [changed from Drama PV; note that the Japanese is already just ドラマ for Drama PV]
- Music PV (音楽PV)
- Other (その他)
